### PR TITLE
ENH add conda-forge.yml cleanup migrator

### LINKED ISSUE #770: Cleanup `max_py_ver` and `max_r_ver`
These are not used anymore.

### PR DESCRIPTION
**Pull request by beckermr**
Closes #770

### CODE REVIEW

**Comment by beckermr** [comment]
OK @CJ-Wright @isuruf this one is ready to go!

File: conda_forge_tick/migrators/max_ver.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            yaml = YAML()
+            yaml.indent(mapping=2, sequence=4, offset=2)
+
+            with open(cfg_path, 'r') as fp:
+                cfg = yaml.load(fp.read())
+
+            if 'max_r_ver' in cfg:
+                del cfg['max_r_ver']
+            if 'max_py_ver' in cfg:
+                del cfg['max_py_ver']
```
Comment (isuruf) [inline conda_forge_tick/migrators/max_ver.py]: We should also delete `compiler_stack`

Comment (beckermr) [reply conda_forge_tick/migrators/max_ver.py]: Ugh ok. Let me generalize the Migrator to do generic conda-forge.yml cleanup I guess.

Comment (beckermr) [reply conda_forge_tick/migrators/max_ver.py]: done!

**Comment by beckermr** [comment]
OK @isuruf and @CJ-Wright this is ready for another look!

**Comment by beckermr** [comment]
@isuruf @scopatz We good to merge?

File: conda_forge_tick/migrators/conda_forge_yaml_cleanup.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+from conda_forge_tick.xonsh_utils import indir
+from conda_forge_tick.migrators.core import MiniMigrator
+
+if typing.TYPE_CHECKING:
+    from ..migrators_types import AttrsTypedDict
+
+
+class CondaForgeYAMLCleanup(MiniMigrator):
+    keys_to_remove = ['max_r_ver', 'max_py_ver', 'compiler_stack']
```
Comment (isuruf) [inline conda_forge_tick/migrators/conda_forge_yaml_cleanup.py]: Can you also add `min_py_ver` and `min_r_ver`?

Comment (beckermr) [reply conda_forge_tick/migrators/conda_forge_yaml_cleanup.py]: Sure!

**Comment by beckermr** [comment]
Weird test failure. Will fix tonight and then merge.

**Pull request merged**